src/bmi_cfe.c

```diff
@@ -1714,10 +1736,61 @@ static int Get_value_ptr (Bmi *self, const char *name, void **dest)
 
 /**************************************************************************************************/
     
-    
-    //NOT MESSING WITH nash_n (number of nash cascades) cause it is a bit of a side
-    //effecting value when it changes
+    if (strcmp (name, "satpsi") == 0) {
+        cfe_state_struct *cfe_ptr;
+        cfe_ptr = (cfe_state_struct *) self->data;
+        *dest = (void*)&cfe_ptr->NWM_soil_params.satpsi;
+        return BMI_SUCCESS;
+    }    
 
+    if (strcmp (name, "wltsmc") == 0) {
+        cfe_state_struct *cfe_ptr;
+        cfe_ptr = (cfe_state_struct *) self->data;
+        *dest = (void*)&cfe_ptr->NWM_soil_params.wltsmc;
+        return BMI_SUCCESS;
+    } 
+
+    if (strcmp (name, "alpha_fc") == 0) {
+        cfe_state_struct *cfe_ptr;
+        cfe_ptr = (cfe_state_struct *) self->data;
+        *dest = (void*)&cfe_ptr->NWM_soil_params.alpha_fc;
+        return BMI_SUCCESS;
+    }  
+    if (strcmp (name, "refkdt") == 0) {
+        cfe_state_struct *cfe_ptr;
+        cfe_ptr = (cfe_state_struct *) self->data;
+        *dest = (void*)&cfe_ptr->NWM_soil_params.refkdt;
+        return BMI_SUCCESS;
+    }    
+     if (strcmp (name, "a_Xinanjiang_inflection_point_parameter") == 0) {
+        cfe_state_struct *cfe_ptr;
+        cfe_ptr = (cfe_state_struct *) self->data;
+        *dest = (void*)&cfe_ptr->direct_runoff_params_struct.a_Xinanjiang_inflection_point_parameter;
+        return BMI_SUCCESS;
+    }  
+
+     if (strcmp (name, "b_Xinanjiang_shape_parameter") == 0) {
+        cfe_state_struct *cfe_ptr;
+        cfe_ptr = (cfe_state_struct *) self->data;
+        *dest = (void*)&cfe_ptr->direct_runoff_params_struct.b_Xinanjiang_shape_parameter;
+        return BMI_SUCCESS;
+    }  
+
+     if (strcmp (name, "x_Xinanjiang_shape_parameter") == 0) {
+        cfe_state_struct *cfe_ptr;
+        cfe_ptr = (cfe_state_struct *) self->data;
+        *dest = (void*)&cfe_ptr->direct_runoff_params_struct.x_Xinanjiang_shape_parameter;
+        return BMI_SUCCESS;
+    } 
+             
+     if (strcmp (name, "N_nash") == 0) {
```

Comment:
Again, `N_nash` is a variable that, if changed, requires reallocation of nash storage array
```
 model->nash_storage = malloc(sizeof(double) * model->N_nash);
        for (j = 0; j < model->N_nash; j++)
            model->nash_storage[j] = 0.0;
    }
```
which is set at initialization.  So if this variable changes, the nash_storage pointer needs to be reallocated to the correct size.